src/address/normalize-address.ts

```diff
@@ -0,0 +1,65 @@
+import { EventEmitter } from "../isomorphic.node";
+import { NormalizedConfig } from "../config";
+import {
+  AddressValidateParams,
+  AddressValidateResult,
+  callJsonRpcMethod,
+} from "../json-rpc";
+import { formatAddress } from "./format-address";
+import {
+  Address,
+  AddressValidationResult,
+  NormalizedAddress,
+} from "./public-types";
+import { validateAddress } from "./validate-address";
+import { validateInputAddress } from "./validate-input-address";
+import { ShipEngineError } from "../errors";
+import { ErrorCode, ErrorType } from "../enums";
+
+/**
+ * Normalizes an address and returns it.
+ */
+export async function normalizeAddress(
+  address: Address,
+  config: NormalizedConfig,
+  events: EventEmitter
+  // @ts-expect-error TypeScript is confused by the return in the for loop
+): Promise<NormalizedAddress | ShipEngineError> {
+  let error: string;
+
+  try {
+    const result: AddressValidationResult = await validateAddress(
+      address,
+      config,
+      events
+    );
+    if (result.normalizedAddress) {
```

Comment:
Why are you checking for `normalizedAddress` here?  It's already checked in the `doesNormalizedAddressHaveErrors()` function.

Reply:
TS was giving me errors on line 38: 

` return result.normalizedAddress;` 

saying that the value might not be populated. I will remove it and include the exact message.

Reply:
My IDE was giving me this error, so I checked for its presence first. 

TS2322: Type 'NormalizedAddress | undefined' is not assignable to type 'NormalizedAddress'. Type 'undefined' is not assignable to type 'NormalizedAddress'. 

Suggestion: Make 'normalizeAddress' return 'Promise<NormalizedAddress | undefined>

Reply:
The app still runs and the linter doesn't complain. Maybe this is a TS rule that we don't care about enforcing?

Reply:
I can't resolve this one. Can we pair on this?